zebra-state/src/service.rs

```diff
@@ -270,6 +270,95 @@ impl StateService {
             state: IterState::NonFinalized(hash),
         }
     }
+
+    /// Return an iterator over the finalized blocks of the chain
+    /// starting (and descending) from the block identified by `height`.
+    ///
+    /// The block identified by `height` is included in the chain of blocks yielded
+    /// by the iterator.
+    pub fn chain_finalized(&self, height: block::Height) -> Iter<'_> {
+        Iter {
+            service: self,
+            state: IterState::Finalized(height),
+        }
+    }
+
+    /// Return a list of blocks ahead given a list of blocks.
+    ///
+    /// 1. Make sure we have the list of blocks provided in our chain.
+    /// 2. Get a max of 500 hashes ahead of the provided tip.
+    pub fn find_chain_hashes(
+        &self,
+        block_locator: Vec<block::Hash>,
+        stop: Option<block::Hash>,
+    ) -> Option<Vec<block::Hash>> {
+        let locator_tip_hash = block_locator.first()?;
+        let locator_tip_height = self.height_by_hash(*locator_tip_hash)?;
+        let chain = self.chain_finalized(locator_tip_height);
+
+        tracing::info!(
+            "REQUESTED TIP: {:?} {:?}",
+            locator_tip_height,
+            locator_tip_hash
+        );
+
+        // Block locator are not continuous but we need to make sure all the blocks
+        // provided are in our chain and in the right order.
+        let mut collect_block_locator: Vec<block::Hash> = Vec::new();
+        let mut index = 0;
+        for block in chain {
+            // We get out of the loop as soon as we collect a vector of equal
+            // lenght of the provided.
+            if collect_block_locator.len() == block_locator.len() {
+                break;
+            }
+
+            if block.hash() == block_locator[index] {
+                collect_block_locator.push(block.hash());
+                index += 1
+            }
+        }
+
+        // If the 2 vectors are not equal we cant continue.
+        if collect_block_locator != block_locator {
+            return None;
+        }
+
+        // The number of next hashes to return.
+        let max_results = 500;
```

Comment:
This is a constant, so let's define it using `const`?
```suggestion
        const MAX_FIND_BLOCK_HASHES_RESULTS: usize = 500;
```